### DISCUSSION TITLE
Can we set the workspace timeout property in template?

### DISCUSSION THREAD

**Discussion by mm00922318** [coder/coder#6045] [General] [answered]
By default the workspaces are coming set as auto-stop after 4 hours. Can we change this value or remove this timeout using template?

**Comment by matifali** [comment, marked as answer]
For now, you can change this setting from the dashboard UI.
Set the `Default auto stop` and `Max life time` to `0`.
![image](https://github.com/coder/coder/assets/10648092/9f2b0770-5e48-4418-928c-5cab6cf8e75c)

**Reply by samkeet** [reply]
@matifali Assuming I have set autostop hours to 8.
1. If I don't use my workspaces for 8 hours from creation, would it be stopped at the end of 8 hours or sometime before?
2. Even if there is a single instance where I connect to it, that should extend the duration by 1 hour (default). Is that correct?

My confusion is when does the process to determine any active connection - run?
1. At the end of first 8 hours?
2. Then at the end of every 1 hour?

**Reply by matifali** [reply]
> If I don't use my workspaces for 8 hours from creation, would it be stopped at the end of 8 hours or sometime before?

Yes.

>Even if there is a single instance where I connect to it, that should extend the duration by 1 hour (default). Is that correct?

If you connect within the last hour (7+), then you will get an activity bump. Otherwise, the workrpsace will autostop as per the configured autostop time, ie. 8 hours. 

@stirby for vetting my claim.

**Reply by samkeet** [reply]
Thank you!
